tests/integration/states/archive.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+'''
+Tests for the file state
+'''
+# Import python libs
+from __future__ import absolute_import
+import os
+import shutil
+
+# Import Salt Testing libs
+from salttesting.helpers import ensure_in_syspath
+ensure_in_syspath('../../')
+
+# Import salt libs
+import integration
+import salt.utils
+
+STATE_DIR = os.path.join(integration.FILES, 'file', 'base')
+ARCHIVE_DIR = '/tmp/archive/'
+ARCHIVE_TAR_SOURCE = 'https://github.com/downloads/Graylog2/'\
```

Comment:
I think this is backed by `file.managed` which means we can use a `salt://` URI. I'd much rather have the master deliver this file then depend on an external source. Perhaps test with that approach? (I could be totally wrong here. Going on memory.)